src/Query/QueryBuilder.php

```diff
@@ -335,6 +335,8 @@ public function executeStatement(): int|string
      * </code>
      *
      * @return string The SQL query string.
+     *
+     * @throws Exception
```

Comment:
This is too generic. `QueryBuilder` should only throw `QueryException`s.

The fact that it also throws a more generic `Exception` because it may try to connect to the database to detect the database platform is a design flaw. If this exception is reported as unhandled somewhere in a static analysis report, I'd rather suppress it than document in the API.

Reply:
Thank you for reviewing!
Changed to the more specific class, feel free to close this PR if it's not necessary.

To fix the PHP Stan, i've added an inline doc tag, as in https://github.com/pimcore/customer-data-framework/pull/580